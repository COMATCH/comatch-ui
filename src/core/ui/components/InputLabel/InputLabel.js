import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './InputLabel.scss';

const propTypes = {
    /**
     * Additional class names
     */
    className: PropTypes.string,
    /**
     * HTML for attribute
     */
    htmlFor: PropTypes.string,
    /**
     * HTML id attribute
     */
    id: PropTypes.string,
    required: PropTypes.bool,
    /**
     * If true will use `<span>` instead of `<label>`
     */
    span: PropTypes.bool,
    /**
     * Text content of `Button`
     */
    text: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,
};

const defaultProps = {
    className: null,
    htmlFor: null,
    id: null,
    required: false,
    span: false,
};

/**
 * A label element to be used with inputs. By default uses a `<label>` tag,
 * but can also be a `<span>` if the `span` prop is set to true.
 */

export const InputLabel = ({ className, htmlFor, id, required, span, text }) =>
    span ? (
        <span id={id} className={classNames('InputLabel', className)}>
            {text}
            {required && '*'}
        </span>
    ) : (
        <label id={id} className={classNames('InputLabel', className)} htmlFor={htmlFor}>
            {text}
            {required && '*'}
        </label>
    );

InputLabel.defaultProps = defaultProps;

InputLabel.propTypes = propTypes;
