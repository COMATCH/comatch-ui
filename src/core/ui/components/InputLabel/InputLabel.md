
## InputLabel

From [`InputLabel.js`](InputLabel.js)

A label element to be used with inputs. By default uses a `<label>` tag,
but can also be a `<span>` if the `span` prop is set to true.

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**className** | `String` | `null` | :x: | Additional class names
**htmlFor** | `String` | `null` | :x: | HTML for attribute
**id** | `String` | `null` | :x: | HTML id attribute
**required** | `Boolean` | `false` | :x: | 
**span** | `Boolean` | `false` | :x: | If true will use `<span>` instead of `<label>`
**text** | `Union<ReactNode \| String>` |  | :white_check_mark: | Text content of `Button`



