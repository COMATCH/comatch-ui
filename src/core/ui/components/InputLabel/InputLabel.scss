@import '../../../styles/variables.scss';

// Label element for Input components
// Common styles regardless of html tag
.InputLabel {
    color: $co-color-gray;
    font-size: $co-font-xs;
    line-height: $co-line-height-xs;
    font-weight: bold;
    margin-bottom: 5px;
    transition: color 250ms ease-out;
    &[for] {
        cursor: pointer;
    }
}

// Styles for <label> tag
label.InputLabel {
    display: block;
    font-size: $co-font-xs;
}

// Styles for <span> tag
span.InputLabel {
    font-size: $co-font-sm;
}

*.has-error .InputLabel {
    // All .InputLabel elements contained
    // within an erroneous Input component.
    color: $co-color-red;
}
