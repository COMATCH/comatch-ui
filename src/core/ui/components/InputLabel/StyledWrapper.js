import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { palette, typography } from '../../../styles/variables';

const generateLabelStyling = ({ isLabel }) => !isLabel ? '' : `
    display: block;
    font-size: ${typography.fontXs};
`;

const StyledWrapper = styled(forwardRef(({ isLabel, ...rest }, ref) => <span {...rest} ref={ref} />))`
    color: ${palette.gray};
    font-size: ${typography.fontSm};
    line-height: ${typography.lineHeightXs};
    font-weight: bold;
    margin-bottom: 5px;
    transition: color 250ms ease-out;
    &[for] {
        cursor: pointer;
    }

    *.has-error & {
      color: ${palette.red};
    }

    ${generateLabelStyling}
`;

export { StyledWrapper };
