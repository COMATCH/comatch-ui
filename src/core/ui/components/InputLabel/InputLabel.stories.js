import React from 'react';
import { storiesOf } from '@storybook/react';
import { InputLabel } from './InputLabel';

storiesOf('InputLabel', module).add('with basic style', () => (
    <>
        <InputLabel text="Sample label" />
        <code>{'<InputLabel text="Sample label" />'}</code>
    </>
));
