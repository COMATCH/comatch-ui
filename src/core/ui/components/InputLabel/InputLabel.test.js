import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { InputLabel } from './InputLabel';
import '../../../config/tests/setup';

describe('InputLabel', () => {
    const text = 'A very random text';
    it('should render correctly', () => {
        const inputLabel = renderer.create(<InputLabel  text={text} />).toJSON();
        expect(inputLabel).toMatchSnapshot();
    });
    it('should render InputLabel as a default class', () => {
        const item = shallow(<InputLabel text={text}/>);
        expect(item.hasClass('InputLabel')).toEqual(true);
    });
    it('should render InputLabel with extra className if passed by the props', () => {
        const className = 'A very random className';
        const item = shallow(<InputLabel text={text} className={className} />);
        expect(item.hasClass(className)).toEqual(true);
    });
    it('should have asterisk after the text if the required prop is set to true', () => {
        const item = shallow(<InputLabel text={text} required />);
        expect(item.find('label').text()).toEqual(`${text}*`);
    });
    it('should render a <span> instead of <label> if span prop is set to true', () => {
        const item = shallow(<InputLabel text={text} span />);
        expect(item.find('label').length).toEqual(0);
        expect(item.find('span').length).toEqual(1);
    });
    it('should set an id attribute if the id prop is given', () => {
        const id = 'A very random id';
        const item = shallow(<InputLabel text={text} id={id} />);
        expect(item.prop('id')).toEqual(id);
    });
});