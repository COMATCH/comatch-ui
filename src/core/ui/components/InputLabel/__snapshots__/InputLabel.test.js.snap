// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel should render correctly 1`] = `
<label
  className="InputLabel"
  htmlFor={null}
  id={null}
/>
`;

exports[`InputLabel snapshot of span prop set to true 1`] = `
<span
  className="InputLabel"
  id={null}
/>
`;
